### PR TITLE
Address review comments

### PR DESCRIPTION
**Pull request by aristotelos**
Revive https://github.com/package-url/purl-spec/pull/245 by adding the following changes:

- `registry_url` -> `repository_url`
- `registry_version` -> `port_revision`
- Remove percent escaping from `repository_url` to be more consistent with other uses of `repository_url` in the purl spec

### CODE REVIEW

**Comment by aristotelos** [comment]
Hi @michaelbprice, because security is important I am trying to revive your https://github.com/package-url/purl-spec/pull/245 by addressing some open review comments. I see you didn't yet respond to my comments. I hope you have some time to review this PR and if it is OK to you, merge it, so that your PR can be moved forward.

**Comment by michaelbprice** [comment]
Currently very busy with CppCon. I'll take a look at your comment in my PR and this one that will update it when I get a chance. Feel free to ping me on it next week if I don't get to it by this weekend.

**Comment by aristotelos** [comment]
Do take the time to enjoy CppCon!

**Comment by aristotelos** [comment]
> Feel free to ping me on it next week if I don't get to it by this weekend.

@michaelbprice Ping! 😄

**Comment by aristotelos** [comment]
@michaelbprice Ping 2!

File: PURL-TYPES.rst [hunk 1]
```diff
@@ -530,32 +530,21 @@ vcpkg
...
-    - For filesystem registries, an string that can be used to identify specific versions, such as Coordinated Universal Time (UTC) as described in `RFC 3339 <https://datatracker.ietf.org/doc/html/rfc3339>`_. If using a UTC time, be sure to percent-encode it correctly. **Optional**
-
-  - Additional qualifiers may be present on the purl that provide additional information about the context in which the package is being used, such as build configuration or platform information. These additional qualifiers must be tolerated during parsing and can be ignored if the parser does not expect them.
-
+- ``version``: The upstream version of the port (excluding the port file revision). **Required**
 - ``subpath``: Currently reserved for future use and should be empty.
+- Qualifier ``repository_url``: The vcpkg registry where the package is available (optional). If omitted, ``https://github.com/microsoft/vcpkg`` as default registry is assumed.
+- Qualifier ``port_revision``: A string specifying the [port file revision](https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json#port-version) (optional).
```
Comment (michaelbprice) [inline PURL-TYPES.rst]:
```suggestion
- Qualifier ``port_revision``: A string specifying the [port file revision](https://learn.microsoft.com/vcpkg/reference/vcpkg-json#port-version) (optional).
```

**Comment by michaelbprice** [comment]
I'm still not convinced that percent-encoding the registry URL is "wrong", but I can live with removing that language. As was pointed out on the PR into the upstream, percent encoding could theoretically still be done, and parsers would need to deal with it. The situations where it would be absolutely needed (a registry URL that has qualifiers itself separated by ampersands) are unlikely to occur.

**Comment by michaelbprice** [comment]
I'd still like to retain the `registry_version` qualifier, but we have language that still gives us some flexibility going forward. The reason why I think it is crucial is that along with the `repository_url` it allows you to narrow down to a specific point-in-time. Otherwise, it's possible that a PURL could not be traced back to the real version of the registry that was used when the PURL was generated. The simplest case I could think of would be that a user makes a mistake and changes a port in a way that is meaningful but doesn't update the port version.

Without that snapshot in time, I'm worried that accidents will happen that will cause confusion or even that a malicious user could wreak havoc in some way that I haven't yet foreseen. But with that commit SHA there, if something has changed, at least someone investigating and tracing backwards could see that something has gone wrong.

**Comment by aristotelos** [comment]
> I'd still like to retain the `registry_version` qualifier, but we have language that still gives us some flexibility going forward. The reason why I think it is crucial is that along with the `repository_url` it allows you to narrow down to a specific point-in-time. Otherwise, it's possible that a PURL could not be traced back to the real version of the registry that was used when the PURL was generated. The simplest case I could think of would be that a user makes a mistake and changes a port in a way that is meaningful but doesn't update the port version.
> 
> Without that snapshot in time, I'm worried that accidents will happen that will cause confusion or even that a malicious user could wreak havoc in some way that I haven't yet foreseen. But with that commit SHA there, if something has changed, at least someone investigating and tracing backwards could see that something has gone wrong.

Good point, I have added a `registry_revision` back for online registries. I however think it is not relevant for filesystem registries or overlay ports because those are not online and therefore traceability will not help. Filesystem registries are an edge case anyways so I think adding additional complexity to the purl spec for vcpkg for that case is not the best thing to do.

I saw I also did not describe the filesystem registry or overlay port concept yet, so added that back.

Also added a full example with additional, unspecified qualifiers.

**Comment by michaelbprice** [comment]
LGTM. Let's go! Hopefully this revives the upstream PR. They've got quite a backlog though.

**Pull request merged**